src/client/datascience/jupyter/kernels/cellExecution.ts

```diff
@@ -766,8 +812,66 @@ export class CellExecution {
             await this.handleExecuteReply(msg, clearState);
 
             // Set execution count, all messages should have it
-            if ('execution_count' in msg.content && typeof msg.content.execution_count === 'number') {
-                await updateCellExecutionCount(this.cell, msg.content.execution_count);
+            if ('execution_count' in msg.content && typeof msg.content.execution_count === 'number' && this.task) {
+                this.task.executionOrder = msg.content.execution_count;
+            }
+        }
+    }
+    /**
+     * Execution of Cell B could result in updates to output in Cell A.
+     */
+    private async handleUpdateDisplayDataMessage(msg: KernelMessage.IUpdateDisplayDataMsg): Promise<void> {
+        const document = this.cell.notebook;
+        // Find any cells that have this same display_id
+        for (const cell of document.cells) {
+            if (cell.kind !== NotebookCellKind.Code) {
+                continue;
+            }
+
+            // Find the cell output that needs ot be updated.
+            const outputToBeUpdated = cell.outputs.find((cellOutput) => {
+                const output = translateCellDisplayOutput(cellOutput);
+                if (
+                    (output.output_type === 'display_data' || output.output_type === 'execute_result') &&
+                    output.transient &&
+                    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                    (output.transient as any).display_id === msg.content.transient.display_id
+                ) {
+                    return true;
+                } else {
+                    return false;
+                }
+            });
+            if (outputToBeUpdated) {
+                const output = translateCellDisplayOutput(outputToBeUpdated);
+                const newOutput = cellOutputToVSCCellOutput({
+                    ...output,
+                    data: msg.content.data,
+                    metadata: msg.content.metadata
+                });
+                // If there was no output and still no output, then nothing to do.
+                if (outputToBeUpdated.outputs.length === 0 && newOutput.outputs.length === 0) {
+                    return;
+                }
+                // Compare outputs (at the end of the day everything is serializable).
+                // Hence this is a safe comparison.
+                if (cell.outputs.length === newOutput.outputs.length && fastDeepEqual(cell.outputs, newOutput)) {
```

Comment:
Cell outputs can be huge. Could this equality compare be a perf issue? Say you are dragging around a widget slider that's updating a complex chart in another cell. It's going to be doing a equality comparison on the entire output each drag step.

Reply:
Well, all of the output is already in memory, hence this is basically a comparison of in memory objects (arrays).
So, i don't see much of an issue here. 
I've optimized the comparison, to check each item individually, if something is different, then it will break early.